### PR TITLE
Improve the test coverage

### PR DESCRIPTION
**Pull request by kalaiyarasiganeshalingam**
## Purpose
$Subject

This pr depends on [#176](https://github.com/ballerina-platform/module-ballerina-task/pull/176)

Related issue: [#1425](https://github.com/ballerina-platform/ballerina-standard-library/issues/1425)
## Checklist
- [ ] Linked to an issue
- [ ] Updated the changelog
- [ ] Added tests

### CODE REVIEW

File: cache-ballerina/tests/test.bal [hunk 1]
```diff
@@ -469,3 +469,48 @@ isolated function testCreateCacheWithNegativeDefaultMaxAge() {
...
+        test:assertEquals(result.toString(), "error Error (\"Unsupported cache value '()' for the key: A.\")");
+    } else {
+         test:assertFail("Output mismatched");
+    }
+}
+
+@test:Config {
+    groups: ["cache", "invalidate", "negative"]
+}
+isolated function testInvalidateWithNullValue() {
```
Comment (ldclakmal) [inline cache-ballerina/tests/test.bal:507]: Shall we rename the test case as follows: 
```suggestion
isolated function testInvalidateWithNonExistingValue() {
```

**Review by ldclakmal — COMMENTED** [review]
LGTM

File: cache-ballerina/tests/test.bal [hunk 2]
```diff
@@ -469,3 +469,48 @@ isolated function testCreateCacheWithNegativeDefaultMaxAge() {
...
 }
+
+@test:Config {
+    groups: ["cache", "cleanup", "negative"]
+}
+isolated function testCreateCacheWithNegativeCleanUpInterval() {
+    CacheConfig config = {
+        cleanupInterval: -1d
+    };
+    Cache|error cache = trap new(config);
```
Comment (shafreenAnfar) [inline cache-ballerina/tests/test.bal:480]: Do we need trap here ?

Comment (kalaiyarasiganeshalingam) [reply cache-ballerina/tests/test.bal:480]: Yes, It needs to capture the panic as we are using panic in the int().

Comment (shafreenAnfar) [reply cache-ballerina/tests/test.bal:480]: I see, can we capture the panic and return it as an error ?

Comment (kalaiyarasiganeshalingam) [reply cache-ballerina/tests/test.bal:480]: Yes, we can but this will change the API. Can we change it now?

Comment (kalaiyarasiganeshalingam) [reply cache-ballerina/tests/test.bal:480]: Yes, we can but this will change the API. Can we change it now?

Comment (daneshk) [reply cache-ballerina/tests/test.bal:480]: We need to look into the possibility of returning an error. @kalaiyarasiganeshalingam can you create an issue for this.

**Review by shafreenAnfar — APPROVED** [review]

**Review by daneshk — APPROVED** [review]

File: cache-ballerina/tests/test.bal (hunk 2 above)
Comment (kalaiyarasiganeshalingam) [reply cache-ballerina/tests/test.bal:480]: Please find the issue for this.
https://github.com/ballerina-platform/ballerina-standard-library/issues/1435

**Pull request merged**